libsel4vm/src/arch/arm/vgic/vgic.c

```diff
@@ -195,8 +195,13 @@ static inline size_t LR_OF_NEXT(size_t idx)
 {
     struct lr_of *dummy_lr_of; /* need this to use ARRAY_SIZE() */
     const size_t idx_max = ARRAY_SIZE(dummy_lr_of->irqs) - 1;
-    assert(idx <= idx_max); /* the index must be sane when calling this */
-    return (idx == idx_max) ? 0 : (idx + 1);
+    /* The index must be sane when calling this, otherwise something seems
+     * broken. In debug builds the assert will stop execution to highlight the
+     * problem. In release builds it's handled gracefully and the index starts
+     * at 0 again - that's be best we can do here.
+     */
+    assert(idx <= idx_max);
+    return (idx <= idx_max) ? (idx + 1) : 0;
```

Comment:
If you want to return `idx + 1`, only when it would be a valid index into the array, you should be doing `idx < idx_max` right?

Reply:
Yes